homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,240 @@
+            raise Unresolvable("Diskstation not initialized")
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=None,
+            media_class=MediaClass.DIRECTORY,
+            media_content_type=MediaClass.IMAGE,
+            title="Synology Photos",
+            can_play=False,
+            can_expand=True,
+            children_media_class=MediaClass.DIRECTORY,
+            children=[
+                *await self._async_build_diskstations(item),
+            ],
+        )
+
+    async def _async_build_diskstations(
+        self, item: MediaSourceItem
+    ) -> list[BrowseMediaSource]:
+        """Handle browsing different diskstations."""
+        ret = []
+
+        if not item.identifier:
+            for entry in self.entries:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=entry.unique_id,
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=f"{entry.title} - {entry.unique_id}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+            return ret
+        identifier = SynologyPhotosMediaSourceIdentifier(item.identifier)
+        diskstation: SynologyDSMData = self.hass.data[DOMAIN][identifier.unique_id]
+
+        if identifier.album_id is None:
+            # Get Albums
+            try:
+                albums = await diskstation.api.photos.get_albums()
+            except SynologyDSMException:
+                return []
+
+            ret = [
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            ]
+            for album in albums:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{item.identifier}/{album.album_id}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=album.name,
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+            return ret
+
+        # Request items of album
+        # Get Items
+        album = SynoPhotosAlbum(int(identifier.album_id), "", 0)
+        try:
+            album_items = await diskstation.api.photos.get_items_from_album(
+                album, 0, 1000
+            )
+        except SynologyDSMException:
+            return []
+
+        ret = []
+        for items_item in album_items:
```

Comment:
```suggestion
        for album_item in album_items:
```
we should name it album_item